cmd/mailbox-controller/controller.go

```diff
@@ -202,105 +191,116 @@ func (ctl *mbCtl) sync1(ctx context.Context, ref any) {
+		mbsName, err = ctl.mbsNameOfSynctarget(st)
 		if err != nil {
-			return true
+			if err.Error() == errNoSpaceId.Error() {
+				logger.V(4).Info("Failed to retrieve Space ID. Retrying", "syncTargetName", st.Name)
+				return true
+			} else {
+				logger.Error(err, "Failed to construct mailbox space name from SyncTarget", "syncTargetName", st.Name)
+				return false
+			}
 		}
-		ctl.queue.Add(mbwsName)
-		return false
 	}
-	mbwsName, ok := refany.(string)
-	if !ok {
-		logger.Error(nil, "Sync expected a string", "ref", refany, "type", fmt.Sprintf("%T", refany))
-		return false
+	if mbsName == "" {
+		mbsName, ok = refany.(string)
+		if !ok {
+			logger.Error(nil, "Sync expected a string", "ref", refany, "type", fmt.Sprintf("%T", refany))
+			return false
+		}
 	}
-	parts := strings.Split(mbwsName, wsNameSep)
+	parts := strings.Split(mbsName, mbsNameSep)
 	if len(parts) != 2 {
-		logger.V(3).Info("Ignoring non-mailbox workspace name", "wsName", mbwsName)
+		logger.V(3).Info("Ignoring non-mailbox space name", "spaceName", mbsName)
 		return false
 	}
-	byIndex, err := ctl.syncTargetIndexer.ByIndex(mbwsNameIndexKey, mbwsName)
+	byIndex, err := ctl.syncTargetIndexer.ByIndex(mbsNameIndexKey, mbsName)
 	if err != nil {
-		logger.Error(err, "Failed to lookup SyncTargets by mailbox workspace name", "mbwsName", mbwsName)
+		logger.Error(err, "Failed to lookup SyncTargets by mailbox space name", "mbsName", mbsName)
 		return false
 	}
 	var syncTarget *edgev2alpha1.SyncTarget
 	if len(byIndex) == 0 {
 	} else {
 		syncTarget = byIndex[0].(*edgev2alpha1.SyncTarget)
 		if len(byIndex) > 1 {
-			logger.Error(nil, "Impossible: more than one SyncTarget fetched from index; using the first", "mbwsName", mbwsName, "fetched", byIndex)
+			logger.Error(nil, "Impossible: more than one SyncTarget fetched from index; using the first", "mbsName", mbsName, "fetched", byIndex)
 		}
 	}
-	workspace, err := ctl.workspaceScopedLister.Get(mbwsName)
+	space, err := ctl.spaceLister.Get(mbsName)
 	if err != nil && !k8sapierrors.IsNotFound(err) {
-		logger.Error(err, "Unable to Get referenced Workspace", "mbwsName", mbwsName)
+		logger.Error(err, "Unable to Get referenced space", "mbsName", mbsName)
 		return true
 	}
 	if syncTarget == nil || syncTarget.DeletionTimestamp != nil {
-		if workspace == nil || workspace.DeletionTimestamp != nil {
-			logger.V(3).Info("Both SyncTarget and Workspace are absent or deleting, nothing to do", "mbwsName", mbwsName)
+		if space == nil || space.DeletionTimestamp != nil {
+			logger.V(3).Info("Both SyncTarget and Mailbox space are absent or deleting, nothing to do", "mbsName", mbsName)
 			return false
 		}
-		err := ctl.workspaceScopedClient.Delete(ctx, mbwsName, metav1.DeleteOptions{Preconditions: &metav1.Preconditions{UID: &workspace.UID}})
+		err := ctl.spaceManagementClient.SpaceV1alpha1().Spaces(ctl.spaceProviderNs).Delete(ctx, mbsName, metav1.DeleteOptions{Preconditions: &metav1.Preconditions{UID: &space.UID}})
```

Comment:
I think the code would be shorter if `ctl.spaceManagementClient.SpaceV1alpha1().Spaces(ctl.spaceProviderNs)` were stored in `ctl.spaceClient` at construction time.